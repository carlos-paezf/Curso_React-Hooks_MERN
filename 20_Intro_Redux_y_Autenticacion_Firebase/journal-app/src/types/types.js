/**
 * The above code is creating a constant called types. This constant is an object that has two keys, login and logout. 
 * The values of these keys are strings that represent the actions that can be dispatched. 
 */
export const types = {
    login: '[Auth] login',
    logout: '[Auth] logout',

    uiSetError: '[UI] Set Error',
    uiRemoveError: '[UI] Remove Error'
}