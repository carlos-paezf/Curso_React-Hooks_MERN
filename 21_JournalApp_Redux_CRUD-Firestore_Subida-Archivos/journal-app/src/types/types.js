/**
 * The above code is creating a constant called types. This constant is an object that has two keys, login and logout. 
 * The values of these keys are strings that represent the actions that can be dispatched. 
 */
export const types = {
    login: '[Auth] login',
    logout: '[Auth] logout',

    uiSetError: '[UI] Set Error',
    uiRemoveError: '[UI] Remove Error',
    uiStartLoading: '[UI] Start Loading',
    uiFinishLoading: '[UI] Finish Loading',

    notesAddNew: '[Notes] Add new entry',
    notesActive: '[Notes] Set active note',
    notesLoad: '[Notes] Load notes',
    notesUpdated: '[Notes] Updated note saved',
    notesFileURL: '[Notes] Updated image url',
    notesDelete: '[Notes] Deleted note',
    notesLogoutCleaning: '[Notes] Logout Cleaning'
}