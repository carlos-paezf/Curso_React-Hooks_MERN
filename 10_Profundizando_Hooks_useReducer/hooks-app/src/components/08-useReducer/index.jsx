export { TodoApp } from "./TodoApp";
export { TodoList } from "./TodoList";
export { TodoListItem } from "./TodoListItem";
export { TodoAdd } from "./TodoAdd";