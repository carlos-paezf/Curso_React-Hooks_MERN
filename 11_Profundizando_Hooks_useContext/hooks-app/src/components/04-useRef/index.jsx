export { FocusScreen } from './FocusScreen'
export { RealExampleRef } from './RealExampleRef'