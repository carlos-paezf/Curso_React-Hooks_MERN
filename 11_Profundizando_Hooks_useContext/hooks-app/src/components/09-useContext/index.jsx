export { AppRouter } from "./AppRouter";

export { AboutScreen } from "./AboutScreen";
export { HomeScreen } from "./HomeScreen";
export { LoginScreen } from "./LoginScreen";
export { MainApp } from "./MainApp";

export { NavBar } from "./NavBar";