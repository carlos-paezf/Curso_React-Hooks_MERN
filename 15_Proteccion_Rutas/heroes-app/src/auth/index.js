/* The above code is creating a context object that will be used to share data throughout our
application. */
export { AuthContext } from "./authContext";
export { authReducer } from "./authReducer";