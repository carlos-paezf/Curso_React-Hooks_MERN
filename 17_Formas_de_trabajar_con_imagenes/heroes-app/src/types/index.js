/**
 * The above code is defining the types of actions that can be dispatched. 
 */
export const types = {
    login: '[auth] Login',
    logout: '[auth] Logout'
}