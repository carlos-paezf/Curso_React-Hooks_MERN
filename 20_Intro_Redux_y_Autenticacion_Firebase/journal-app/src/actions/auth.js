import { types } from "../types/types"
import { firebase, googleAuthProvider } from '../firebase/firebase-config'


export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        setTimeout(() => {
            dispatch(login(123, 'Ferrer'))
        }, 3500)
    }
}


export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider)
            .then(({ user: { displayName, uid } }) => dispatch(login(uid, displayName)))
    }
}



export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
        uid,
        displayName
    }
})
