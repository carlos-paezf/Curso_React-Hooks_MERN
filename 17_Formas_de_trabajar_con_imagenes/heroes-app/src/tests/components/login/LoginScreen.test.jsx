import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { AuthContext } from '../../../auth'
import { LoginScreen } from '../../../components/login'
import { types } from '../../../types'

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}))


describe('Test to <LoginScreen /> component', () => {
    const dispatch = jest.fn()

    const contextValue = {
        user: { logged: false },
        dispatch
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialState={['/login']}>
                <LoginScreen />
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('Should match with Snapshot', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('h1').text().trim()).toBe('Login Screen')
    })

    test('Should call login, dispatch, localStorage, navigate', () => {
        const handleClick = wrapper.find('button').prop('onClick')
        handleClick()

        const action = {
            type: types.login,
            payload: {
                name: 'David Ferrer'
            }
        }
        expect(dispatch).toHaveBeenCalledWith(action)
        expect(mockNavigate).toHaveBeenCalledWith('/', { replace: true })
        
        localStorage.setItem('lastPath', '/search?q=Batman')
        handleClick()
        expect(mockNavigate).toHaveBeenCalledWith('/search?q=Batman', { replace: true })
    })
})