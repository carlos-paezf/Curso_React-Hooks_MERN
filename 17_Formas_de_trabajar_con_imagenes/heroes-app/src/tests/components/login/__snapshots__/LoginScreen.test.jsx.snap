// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <LoginScreen /> component Should match with Snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
